cargo-shuttle/src/client.rs

```diff
@@ -5,34 +5,32 @@ use anyhow::{Context, Result};
 use headers::{Authorization, HeaderMapExt};
 use percent_encoding::utf8_percent_encode;
 use reqwest::header::HeaderMap;
-use reqwest::RequestBuilder;
-use reqwest::Response;
+use reqwest::{RequestBuilder, Response};
 use serde::{Deserialize, Serialize};
 use shuttle_common::constants::headers::X_CARGO_SHUTTLE_VERSION;
 use shuttle_common::log::LogsRange;
 use shuttle_common::models::deployment::DeploymentRequest;
 use shuttle_common::models::team;
 use shuttle_common::models::{deployment, project, service, ToJson};
-use shuttle_common::secrets::Secret;
-use shuttle_common::{resource, ApiKey, ApiUrl, LogItem, VersionInfo};
+use shuttle_common::{resource, ApiKey, LogItem, VersionInfo};
 use tokio::net::TcpStream;
 use tokio_tungstenite::tungstenite::client::IntoClientRequest;
 use tokio_tungstenite::{connect_async, MaybeTlsStream, WebSocketStream};
 use tracing::error;
 use uuid::Uuid;
 
 #[derive(Clone)]
-pub struct Client {
-    api_url: ApiUrl,
-    api_key: Option<Secret<ApiKey>>,
+pub struct ShuttleApiClient {
     client: reqwest::Client,
+    api_url: String,
+    api_key: Option<ApiKey>,
+    /// alter behaviour to interact with the new platform
+    beta: bool,
```

Comment:
I would recommend rather creating a new client for beta rather than passing this flag around everywhere

Reply:
The number of beta changes thus far is small enough for me to justify this. We might extract it later, but there will still be many places outside of this client that will do the same thing (modify behaviour based on passed flag).

Reply:
Hmm, that more points to why it would have been a good idea to do it now instead of later when it will be harder.